orca-echo/src/main/java/com/netflix/spinnaker/orca/echo/util/ManualJudgmentAuthzGroupsUtil.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 OpsMx, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.echo.util;
+
+import static java.lang.String.format;
+
+import com.netflix.spinnaker.fiat.model.Authorization;
+import com.netflix.spinnaker.kork.exceptions.SpinnakerException;
+import com.netflix.spinnaker.orca.front50.Front50Service;
+import com.netflix.spinnaker.orca.front50.model.Application;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import retrofit.RetrofitError;
+
+public class ManualJudgmentAuthzGroupsUtil {
+
+  @Autowired(required = false)
+  Front50Service front50Service;
+
+  public ManualJudgmentAuthzGroupsUtil(Optional<Front50Service> front50Service) {
+    this.front50Service = front50Service.orElse(null);
+  }
+
+  public static boolean checkAuthorizedGroups(
```

Comment:
You still have not added any inline comments here around what this function is intended to do.

I don't quite understand why we're checking against application permissions when the manual judgment could not have happened if the user didn't have EXECUTE.

```
@PreAuthorize("hasPermission(this.getPipeline(#id)?.application, 'APPLICATION', 'EXECUTE')")
@RequestMapping(value = "/pipelines/{id}/stages/{stageId}", method = RequestMethod.PATCH)
PipelineExecution updatePipelineStage(
```

Reply:
Added inline comments.

Application permissions has to be 'CREATE, WRITE, EXECUTE' to proceed further in the manual judgment stage.

This method checks if the logged in user has role in the manual judgment stage authorized groups.
   We fetch the user roles and check if that role is authorized in the manual judgment stage role.
   if the user role exists , then we check with the application permission roles.
   If the application permission role has 'READ' then we return false(not authorized)
   If the application permission role has 'CREATE, EXECUTE, WRITE' then we return true(authorized)